ui/pages/confirmations/confirm/stories/signatures/sign-typed-data.stories.tsx

```diff
@@ -0,0 +1,25 @@
+import ConfirmPage from '../../confirm'
+import {
+  ARG_TYPES_SIGNATURE,
+  CONFIRM_PAGE_DECORATOR,
+  SignatureStoryTemplate
+} from './utils';
+import { unapprovedTypedSignMsgV1 } from '../../../../../../test/data/confirmations/typed_sign';
+
+/**
+ * The `<ConfirmPage>` that's displayed when the current confirmation is a version "V1" `eth_signTypedData` signature.
+ */
+export default {
+  title: 'Pages/Confirmation/ConfirmPage/Signatures/SignTypedDataV1',
```

Comment:
Splitting these into separate files makes total sense, but is there a limit to how far we should go?

Arguably `Permit` could go into V3 and V4 for example since it's a subset of that, and the same with SIWE and personal sign but I assume the argument here is we present them as distinct confirmations hence the separate file.

This makes sense to me, just confirming the logic.

Reply:
Good question. I'm up for combining these at this type layer. Will update

Reply:
by distinct confirmations was the initial idea, though with this second thought, distinct confirmations might be tricky to pin since we have layers of distinctions.

Great callout and updated [b19224b](https://github.com/MetaMask/metamask-extension/pull/25283/commits/b19224b8a9210996c09dcec0be2fa0e6fcc8cbb3). We could revert this if we do find it more helpful to later